OpenDreamShared/Dream/Procs/DreamProcOpcode.cs

```diff
@@ -105,43 +107,119 @@ public enum DreamProcOpcodeParameterType {
-        LowerSubjectPronoun = 0x7,      //he, she, they, it
-        UpperPossessiveAdjective = 0x8, //His, Her, Their, Its
-        LowerPossessiveAdjective = 0x9, //his, her, their, its
-        ObjectPronoun = 0xA,            //him, her, them, it
-        ReflexivePronoun = 0xB,         //himself, herself, themself, it
-        UpperPossessivePronoun = 0xC,   //His, Hers, Theirs, Its
-        LowerPossessivePronoun = 0xD,   //his, hers, theirs, its
-
-        Proper = 0xE,                   //String represents a proper noun
-        Improper = 0xF,                 //String represents an improper noun
-
-        OrdinalIndicator = 0x10,        //1st, 2nd, 3rd, 4th, ...
-        PluralSuffix = 0x11,            //-s suffix at the end of a plural noun
-
-        Icon = 0x12,                    //Use an atom's icon
-
-        ColorRed = 0x13,
-        ColorBlue = 0x14,
-        ColorGreen = 0x15,
-        ColorBlack = 0x16,
-        ColorYellow = 0x17,
-        ColorNavy = 0x18,
-        ColorTeal = 0x19,
-        ColorCyan = 0x1A,
-
-        Bold = 0x1B,
-        Italic = 0x1C
+    /// <summary>
+    /// Handles how we write format data into our strings.
+    /// </summary>
+    public static class StringFormatEncoder
+    {
+        /// <summary>
+        /// This is the upper byte of the 2-byte markers we use for storing formatting data within our UTF16 strings.<br/>
+        /// </summary>
+        /// <remarks>
+        /// It is not const because (<see langword="TODO"/>) eventually it would be desirable to make it something else<br/>
+        /// (even though doing so would slightly break parity)<br/>
+        /// because 0xFFxx actually maps to meaningful Unicode code points under UTF16<br/>
+        /// (DM uses this because it uses UTF8 and 0xFF is just an invalid character in that encoding, no biggie)<br/>
+        /// See: "Halfwidth and Fullwidth Forms" on https://en.wikibooks.org/wiki/Unicode/Character_reference/F000-FFFF
+        /// </remarks>
+        public static UInt16 FormatPrefix = 0xFF00;
+
+        /// <summary>
+        /// The lower byte of the aforementioned formatting marker thingies we stuff into our UTF16 strings.<br/>
+        /// To avoid clashing with the (ALREADY ASSIGNED!) 0xFFxx code point space, these values should not exceed 0x005e (94)
+        /// </summary>
+        /// <remarks>
+        /// <see langword="DO NOT CAST TO CHAR!"/> This requires FormatPrefix to be added to it in order to be a useful formatting character!!
+        /// </remarks>
+        public enum FormatSuffix : UInt16
+        {
+            //States that Interpolated values can have (the [] thingies)
+            StringifyWithArticle = 0x0,    //[] and we include an appropriate article for the resulting value, if necessary            
+            StringifyNoArticle = 0x1,      //[] and we never include an article (because it's elsewhere)
+            ReferenceOfValue = 0x2,        //\ref[]
```

Comment:
If `FormatPrefix` was const you could change these lines to
```suggestion
            StringifyWithArticle = FormatPrefix,    //[] and we include an appropriate article for the resulting value, if necessary
            StringifyNoArticle,      //[] and we never include an article (because it's elsewhere)
            ReferenceOfValue,        //\ref[]
```
This would have every value start from `0xFF00` and count up, which could simplify things a good bit.

Reply:
Again, tried to code it for the compiler deciding to do alternative behaviour during its runtime, when reading configuration.